cdap-unit-test/src/test/java/co/cask/cdap/spark/SparkTestRun.java

```diff
@@ -294,6 +302,50 @@ public void testScalaSparkWithObjectStore() throws Exception {
   }
 
   @Test
+  public void testScalaSparkCrossNSStream() throws Exception {
+    getNamespaceAdmin().create(new NamespaceMeta.Builder()
+                                 .setName("streamSpaceForSpark").build());
+    NamespaceId streamSpace = new NamespaceId("streamSpaceForSpark");
+    StreamManager streamManager = getStreamManager(streamSpace.toId(), "testStream");
+    streamManager.createStream();
+    for (int i = 0; i < 50; i++) {
+      streamManager.send(String.valueOf(i));
+    }
+
+    ApplicationManager applicationManager = deploy(SparkAppUsingObjectStore.class);
+    SparkManager sparkManager =
+      applicationManager.getSparkManager(ScalaCrossNSStreamProgram.class.getSimpleName()).start();
+    sparkManager.waitForFinish(1, TimeUnit.MINUTES);
+
+    DataSetManager<KeyValueTable> countManager = getDataset("count");
+    KeyValueTable results = countManager.get();
+    for (int i = 0; i < 50; i++) {
+      byte[] key = String.valueOf(i).getBytes(Charsets.UTF_8);
+      Assert.assertArrayEquals(key, results.read(key));
+    }
+  }
+
+  @Test
+  public void testScalaSparkCrossNSDataset() throws Exception {
+    // Deploy and create a dataset in namespace datasetSpaceForSpark
+    getNamespaceAdmin().create(new NamespaceMeta.Builder()
+                                 .setName("datasetSpaceForSpark").build());
+    NamespaceId dataSpace = new NamespaceId("datasetSpaceForSpark");
+    deploy(dataSpace, SparkAppUsingObjectStore.class);
+    DataSetManager<ObjectStore<String>> keysManager = getDataset(dataSpace.toId(), "keys");
+    prepareInputData(keysManager);
+
+    ApplicationManager applicationManager = deploy(SparkAppUsingObjectStore.class);
+    SparkManager sparkManager =
+      applicationManager.getSparkManager(ScalaCrossNSDatasetProgram.class.getSimpleName()).start();
+    sparkManager.waitForFinish(1, TimeUnit.MINUTES);
+
+    DataSetManager<KeyValueTable> countManager = getDataset("count");
+    checkOutputData(countManager);
```

Comment:
This is an existing method to check the charcount program. On L321, I'm sending different messages to a stream and verify the output dataset. This method won't be able to check that.